templates/rust/pkgs/default.nix

```
@@ -1,6 +1,22 @@
-{rustPlatform}:
+{
+  rustPlatform,
+  # you can add imports here
+}:
 rustPlatform.buildRustPackage rec {
   name = "my-package";
```

```suggestion
  pname = "my-package";
  version = "";
```

As most other builders `buildRustPackage` creates a name by itself from pname and version.

https://github.com/NixOS/nixpkgs/blob/823b0e3c3616af2562335eb3f63801060a1386ee/pkgs/build-support/rust/build-rust-package/default.nix#L18

Nix search relies on these values or is otherwise likely to butcher the name to infer a version.
That thoroughly complicates our search efforts i think
@limeytexan is that a priority?
we can still try to infer versions if we know what language were dealing with by reading the cargo toml or something.json, its however a matter of providing these utils or wrapping the builders (once again)